fall-2020/navigation.py

```diff
@@ -70,24 +66,117 @@ def set_desired_position(self, desired_position):
         Based on the current position tuple, compared to the desired position tuple,
         make decisions on steering, and ensure that the actuations are successful
-
         Returns: True if actuation requests were successful, False if not
         Note: It may be good to notify the user if something unexpected happens!
         """
-        pass
+        #checking if current postion and desired position are same and if same then prompting to input new values and calling navigate again
+        #if both positions are same then we could either take new input and run navigate again like shown below or we could return an error status like the boolean values for actuatuion requests as mentioned above.
+        if self.current_position == self.desired_position :
+            print("TBM is already at desired position or desired position not set,please set new desired position \n")
+            coordinate = input ("please input desired coordinate in the following form: ' x , y , z ' \n")
+            #coordinate value taken at one go and then seperated as inputting the coordinates seperately will take longer time
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
+            return
+        #error checking for if the x given coordinate  is already behind the machine or not
+        if self.current_position(0) > self.desired_position(0): 
+            print("desired position for x axis is behind the TBM ,please put x coordinate which is infront of TBM \n")
+            coordinate = input ("please input desired coordinate in the following form: ' x , y , z ' \n")
+            coordinate.replace(" ","")
+            desired_coordinate_list = coordinate.split(",")
+            desired_position_tuple = ()
+            for number_of_coordinates in range(3):
+                updater_tuple = (desired_coordinate_list[number_of_coordinates],)
+                desired_position_tuple = desired_position_tuple + updater_tuple
+            self.set_desired_position(desired_position_tuple)
+            self.navigate
+            return
+            
+        targetreached = False
+        while targetreached != True :
+            #should add a pausing mechanism to let actuators take input and implement movement before checking new position and distance left to move using time.sleep(.) 
+            #confused about whether the steering methods return true if full movement to desired location complete or only if movement started and movement distance is given
+            #if steering methods return true after their full movement completed ,then loop is not required .
```

Comment:
This is great that you thought about this. so if it were that the actuator methods just initiate movement, then yes you would want to loop through and check the new position, and stop once it reached the desired position. I think for this, we can assume that the steering methods will block until they reach their desired location, and will return true after that. So if you call them you shouldn't have to worry about waiting in this function

Reply:
Thanks, I think for now removing the while loop will be better.